collect_app/src/main/java/org/odk/collect/android/provider/ProjectsProvider.kt

```diff
@@ -0,0 +1,48 @@
+package org.odk.collect.android.provider
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.projects.ProjectsRepository
+import javax.inject.Inject
+
+class ProjectsProvider : ContentProvider() {
+
+    @Inject
+    lateinit var projectsRepository: ProjectsRepository
+
+    // Do not call it in onCreate() https://stackoverflow.com/questions/23521083/inject-database-in-a-contentprovider-with-dagger
+    private fun deferDaggerInit() {
+        DaggerUtils.getComponent(context).inject(this)
+    }
+
+    override fun onCreate(): Boolean {
+        return true
+    }
+
+    override fun query(uri: Uri, projection: Array<out String>?, selection: String?, selectionArgs: Array<out String>?, sortOrder: String?): Cursor? {
+        deferDaggerInit()
+
+        val cursor = MatrixCursor(arrayOf(PROJECT_UUID, PROJECT_NAME))
+        projectsRepository.getAll().forEach {
+            cursor.addRow(arrayOf<Any>(it.uuid, it.name))
+        }
+        return cursor
+    }
+
+    override fun getType(uri: Uri) = ProjectsProviderAPI.CONTENT_TYPE
+
+    override fun insert(uri: Uri, values: ContentValues?) = throw UnsupportedOperationException()
+
+    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<out String>?) = throw UnsupportedOperationException()
+
+    override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<out String>?) = throw UnsupportedOperationException()
+
+    companion object {
```

Comment:
Should these go in `ProjectProviderAPI` seeing as they are part of the "API" of the content provider. I think it would be nice if we could point people to one class to see how to interact with it (it'll be interesting to think about actually documenting some of this).

Reply:
Fixed.